state/protocol/badger/validity.go

```diff
@@ -111,3 +117,150 @@ func validCommit(commit *flow.EpochCommit, setup *flow.EpochSetup) error {
 
 	return nil
 }
+
+type receiptValidator struct {
```

Comment:
I think this new receipt validation code could live in its own `receipt_validator.go` file since it's implementing a distinct component interface. The rest of this file is more helper methods for the larger protocol state.

Reply:
agree

Reply:
Generally, I was wondering about the placement of the interface 
https://github.com/onflow/flow-go/blob/d9d7f00ff2e9304f6c9e19088161ba40a1575123/state/protocol/validation.go#L5-L7
as well as its concrete implementation 
https://github.com/onflow/flow-go/blob/7c3ceeddf342532f0a073f5e2ee2cad5cbb947ce/state/protocol/badger/validity.go#L121-L126

Here are my thoughts:
* The verification components, like `ReceiptValidator` are not specific to the protocol state. There should (or will be) other component which need to verify the validity of Execution Receipts. 
* we have other general service components, such as a [verification component for signatures](https://github.com/onflow/flow-go/blob/7c3ceeddf342532f0a073f5e2ee2cad5cbb947ce/module/verifier.go#L8) in our [`module`](https://github.com/onflow/flow-go/tree/7c3ceeddf342532f0a073f5e2ee2cad5cbb947ce/module) package. Accordingly, we could place the entire content of 
   - interface `ReceiptValidator` into `module/receipt_validator.go`
   - implementation `receiptValidator` into `module/validation/receipt_validator.go`

Reply:
I like it, will do that way if everyone is ok with it.